x-pack/legacy/plugins/siem/server/lib/framework/kibana_framework_adapter.ts

```diff
@@ -21,125 +26,113 @@ import {
+    this.router.get(
+      {
+        path: routePath,
+        validate: { query: configSchema.object({}, { allowUnknowns: true }) },
+        options: {
+          tags: ['access:siem'],
+        },
       },
-      handler: async (request: RequestFacade, h: ResponseToolkit) => {
+      async (context, request, response) => {
         try {
-          const query =
-            request.method === 'post'
-              ? (request.payload as Record<string, any>) // eslint-disable-line @typescript-eslint/no-explicit-any
-              : (request.query as Record<string, any>); // eslint-disable-line @typescript-eslint/no-explicit-any
-
+          const { query } = request;
           const gqlResponse = await runHttpQuery([request], {
-            method: request.method.toUpperCase(),
+            method: 'GET',
             options: (req: RequestFacade) => ({
-              context: { req: wrapRequest(req) },
+              context: { req: wrapRequest(req, context) },
               schema,
             }),
-
             query,
           });
 
-          return h.response(gqlResponse).type('application/json');
+          return response.ok({
+            body: gqlResponse,
+            headers: {
+              'content-type': 'application/json',
+            },
+          });
         } catch (error) {
-          if ('HttpQueryError' !== error.name) {
-            const queryError = Boom.boomify(error);
-
-            queryError.output.payload.message = error.message;
-
-            return queryError;
-          }
-
-          if (error.isGraphQLError === true) {
-            return h
-              .response(error.message)
-              .code(error.statusCode)
-              .type('application/json');
-          }
-
-          const genericError = new Boom(error.message, { statusCode: error.statusCode });
-
-          if (error.headers) {
-            Object.keys(error.headers).forEach(header => {
-              genericError.output.headers[header] = error.headers[header];
-            });
-          }
-
-          // Boom hides the error when status code is 500
-          genericError.output.payload.message = error.message;
-
-          throw genericError;
+          return this.handleError(error, response);
         }
-      },
-      method: ['GET', 'POST'],
-      path: routePath,
-      vhost: undefined,
-    });
+      }
+    );
 
-    if (!this.envMode.prod) {
-      this.server.route({
+    this.router.post(
+      {
+        path: routePath,
+        validate: { body: configSchema.object({}, { allowUnknowns: true }) },
         options: {
           tags: ['access:siem'],
         },
-        handler: async (request: RequestFacade, h: ResponseToolkit) => {
+      },
+      async (context, request, response) => {
+        try {
+          const gqlResponse = await runHttpQuery([request], {
+            method: 'POST',
+            options: (req: RequestFacade) => ({
+              context: { req: wrapRequest(req, context) },
```

Comment:
This was mainly to cut down on the noise in this PR, but if folks think I should pass `context` explicitly as another property I'm happy to make that change.

Reply:
@XavierM

Reply:
#52133